game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritory.java

```diff
@@ -99,30 +103,34 @@ public ProTerritory(final Territory territory) {
     maxUnits = new ArrayList<>(patd.getMaxUnits());
     units = new ArrayList<>(patd.getUnits());
     bombers = new ArrayList<>(patd.getBombers());
-    cantMoveUnits = new ArrayList<>(patd.getCantMoveUnits());
-    maxEnemyUnits = new ArrayList<>(patd.getMaxEnemyUnits());
-    maxEnemyBombardUnits = new HashSet<>(patd.getMaxEnemyBombardUnits());
-    minBattleResult = patd.getMaxBattleResult();
+    maxBattleResult = patd.getMaxBattleResult();
+    value = patd.getValue();
+    seaValue = patd.getSeaValue();
     canHold = patd.isCanHold();
     canAttack = patd.isCanAttack();
     strengthEstimate = patd.getStrengthEstimate();
+
     maxAmphibUnits = new ArrayList<>(patd.getMaxAmphibUnits());
-    maxBombardUnits = new HashSet<>(patd.getMaxBombardUnits());
+    amphibAttackMap = new HashMap<>(patd.getAmphibAttackMap());
+    transportTerritoryMap = new HashMap<>(patd.getTransportTerritoryMap());
     needAmphibUnits = patd.isNeedAmphibUnits();
     strafing = patd.isStrafing();
-    amphibAttackMap = new HashMap<>(patd.getAmphibAttackMap());
     isTransportingMap = new HashMap<>(patd.getIsTransportingMap());
-    transportTerritoryMap = new HashMap<>(patd.getTransportTerritoryMap());
+    maxBombardUnits = new HashSet<>(patd.getMaxBombardUnits());
     bombardOptionsMap = new HashMap<>(patd.getBombardOptionsMap());
     bombardTerritoryMap = new HashMap<>(patd.getBombardTerritoryMap());
+
     currentlyWins = patd.isCurrentlyWins();
     battleResult = patd.getBattleResult();
+
+    cantMoveUnits = new ArrayList<>(patd.getCantMoveUnits());
+    maxEnemyUnits = new ArrayList<>(patd.getMaxEnemyUnits());
+    maxEnemyBombardUnits = new HashSet<>(patd.getMaxEnemyBombardUnits());
     minBattleResult = patd.getMinBattleResult();
     tempUnits = new ArrayList<>(patd.getTempUnits());
     tempAmphibAttackMap = new HashMap<>(patd.getTempAmphibAttackMap());
     loadValue = patd.getLoadValue();
-    value = patd.getValue();
-    seaValue = patd.getSeaValue();
+
```

Comment:
I don't quite see the pattern here, could you elaborate on why you chose to re-arrange the assignments this way?

Reply:
Good question, I re-arranged it to match the field definitions, such that the field definitions and assignments come in the same order. That should make it easier to check the constructors and help avoid bugs.